### PR TITLE
Speed up shellcraft loading and clean up code a bit

### PR DESCRIPTION
**Pull request by zachriggle**
Old:

```
$ time shellcraft
...
shellcraft  0.28s user 0.17s system 98% cpu 0.455 total
```

New:

```
$ time shellcraft
...
shellcraft  0.17s user 0.10s system 102% cpu 0.272 total
```

### CODE REVIEW

File: pwnlib/commandline/shellcraft.py [hunk 1]
```diff
@@ -248,11 +187,11 @@ def main():
     code = pwnlib.asm.asm(code, arch = arch, os = os)
 
     if args.format in ['s', 'str', 'string']:
-        code = _string(code)
+        code = repr(code) + '\n'
```
Comment (IdolfHatler) [inline pwnlib/commandline/shellcraft.py]: This is unfortunately not equivalent.

File: pwnlib/commandline/shellcraft.py [hunk 2]
```diff
@@ -248,11 +187,11 @@ def main():
     code = pwnlib.asm.asm(code, arch = arch, os = os)
 
     if args.format in ['s', 'str', 'string']:
-        code = _string(code)
+        code = '"' + repr(code)[1:-1].replace('"', '\\"')  + '"\n'
```
Comment (IdolfHatler) [inline pwnlib/commandline/shellcraft.py]: It still does not escape `$`.

The only point I can see of `--format string` is if you want it to be put in an `echo -e [foo]` or `echo $'[foo]'`. Right now that would not work.

But perhaps there are other usecases I did not forsee? :)

**Comment by zachriggle** [comment]
Ah, I thought it's supposed to be for a C/Python/Perl-compatible string.  Will fix.

**Comment by IdolfHatler** [comment]
Well I guess that too. I've just never used it for that. :)

**Comment by kokjo** [comment]
Has this been resolved?

**Comment by zachriggle** [comment]
Nope, if forgotten about it. I just need to put the old string print code back in.

**Comment by zachriggle** [comment]
I've rebased this against the current `Gallopsled/pwntools/master` and removed the changes to string-printing per @Idolf's comments

**Pull request merged**